src/Symfony/Component/Notifier/Bridge/Slack/Tests/Block/SlackSectionBlockTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Slack\Tests\Block;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Notifier\Bridge\Slack\Block\SlackImageBlockElement;
+use Symfony\Component\Notifier\Bridge\Slack\Block\SlackSectionBlock;
+
+final class SlackSectionBlockTest extends TestCase
+{
+    public function testCanBeInstantiated(): void
+    {
+        $section = new SlackSectionBlock();
+        $section->text('section text');
+        $section->field('section field');
+        $section->accessory(new SlackImageBlockElement('https://example.com/image.jpg', 'an image'));
+
+        $this->assertSame([
+            'type' => 'section',
+            'text' => [
+                'type' => 'mrkdwn',
+                'text' => 'section text',
+            ],
+            'fields' => [
+                [
+                    'type' => 'mrkdwn',
+                    'text' => 'section field',
+                ],
+            ],
+            'accessory' => [
+                'type' => 'image',
+                'image_url' => 'https://example.com/image.jpg',
+                'alt_text' => 'an image',
+            ],
+        ], $section->toArray());
+    }
+
+    public function testThrowsWhenFieldsLimitReached(): void
+    {
+        $this->expectException(\LogicException::class);
+        $this->expectExceptionMessage('Maximum number of fields should not exceed 10.');
+
+        $section = new SlackSectionBlock();
+        foreach (range(0, 10) as $i) {
+            $section->field($i);
+        }
```

Comment:
This is an interesting way to write a `for` loop. 😃 

I'd like to make sure that the first ten fields are actually accepted. How about this?

```suggestion
        $section = new SlackSectionBlock();
        foreach ($i = 0; 10 > $i; ++$i) {
            $section->field($i);
        }
        
        $this->expectException(\LogicException::class);
        $this->expectExceptionMessage('Maximum number of fields should not exceed 10.');
        $section->field('fail');
```

Reply:
> This is an interesting way to write a `for` loop. 

Don't know what crossed my mind at the time of writing that :smile: updated :)